libtransmission/net.cc

```diff
@@ -516,12 +516,16 @@ std::optional<tr_address> tr_address::from_string(std::string_view address_sv)
 std::string_view tr_address::display_name(char* out, size_t outlen) const
 {
     TR_ASSERT(is_valid());
+    if (!is_valid())
+    {
+        return "Invalid address"sv;
+    }
     return evutil_inet_ntop(tr_ip_protocol_to_af(type), &addr, out, outlen);
 }
 
 [[nodiscard]] std::string tr_address::display_name() const
 {
-    auto buf = std::array<char, INET6_ADDRSTRLEN>{};
+    auto buf = std::array<char, std::max(INET_ADDRSTRLEN, INET6_ADDRSTRLEN) /* just in case */>{};
```

Comment:
When reading the comment outside of this PR, I wouldn't know what it refers to. Are we passing the [required] template parameter just in case?.. Either expand or remove.

Reply:
2348c6e